packages/api/AlvTime.Persistence/Repositories/FlexhourStorage.cs

```diff
@@ -315,16 +316,27 @@ public ObjectResult RegisterPaidOvertime(GenericHourEntry request, int userId)
             _context.PaidOvertime.Add(paidOvertime);
             _context.SaveChanges();
 
-            return new OkObjectResult(paidOvertime);
+            return new PaidOvertimeEntry()
+            {
+                UserId = paidOvertime.Id,
+                Date = paidOvertime.Date,
+                HoursBeforeCompensation = paidOvertime.HoursBeforeCompRate,
+                HoursAfterCompensation = paidOvertime.HoursAfterCompRate
+            };
         }
 
-        return new BadRequestObjectResult("Not enough available hours");

```

Comment:
I like the removal of `BadRequestObjectResult` from the business class. However, when throwing an exception, we will change the behaviour of the API from returning a http status code 400 to http status code 500 - which frontend might handle differently. I guess I wrote a middleware for this purpose, so I guess you can extend it to catch ValidationException and set status code to be BadRequest and return problem details? https://github.com/Alv-no/alvtime/blob/master/packages/api/AlvTimeWebApi/ErrorHandling/ErrorHandlingMiddleware.cs